### PR TITLE
Adding C APIs for crypto opereation HMAC and AES-CCM

### PR DESCRIPTION
**Pull request by abtink**
This PR introduces ot C APIs for crypto operation HMAC and AES-CCM.

### CODE REVIEW

File: src/core/crypto/openthread-crypto.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+#endif
+
+void otCryptoHmacSha256(
+    const uint8_t *aKey, uint16_t aKeyLength,
+    const uint8_t *aBuf, uint16_t aBufLength,
+    uint8_t *aHash)
+{
+    HmacSha256 hmac;
+
+    VerifyOrExit((aKey != NULL) && (aBuf != NULL) && (aHash != NULL), ;);
```
Comment (jwhui) [inline src/core/crypto/openthread-crypto.cpp]: Assert on invalid arguments?

File: src/core/crypto/openthread-crypto.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    const void *aNonce, uint8_t aNonceLength,
+    const void *aHeader, uint32_t aHeaderLength,
+    void *aPlainText, void *aCipherText, uint32_t aLength, bool aEncrypt,
+    void *aTag)
+{
+    AesCcm aesCcm;
+    uint8_t tagLength;
+
+    VerifyOrExit((aKey != NULL) && (aNonce != NULL) && (aHeader != NULL) && (aPlainText != NULL) &&
+                 (aCipherText != NULL) && (aTag != NULL), ;);
```
Comment (jwhui) [inline src/core/crypto/openthread-crypto.cpp]: Assert on invalid arguments?

File: src/core/crypto/openthread-crypto.cpp (hunk 1 above)
Comment (abtink) [inline src/core/crypto/openthread-crypto.cpp]: Yes. I agree. Asserting here would be better... Will change this.

File: src/core/crypto/openthread-crypto.cpp (hunk 2 above)
Comment (abtink) [inline src/core/crypto/openthread-crypto.cpp]: Yes. Will change this...

**Pull request merged**